src/extension/providers/nostr/index.ts

```diff
@@ -19,9 +19,13 @@ export default class NostrProvider {
     if (this.enabled) {
       return { enabled: true };
     }
-    return await this.execute("enable");
+    return await this.execute("enable").then((result) => {
+      if (typeof result.enabled === "boolean") {
+        this.enabled = result.enabled;
+      }
+      return result;
+    });
```

Comment:
what's the best practice? is mixing await and then ok? 

or is something like this better?

```
const result = await this.execute("enable");
if (typeof result.enabled === "boolean") {
  this.enabled = result.enabled;
}
return result;
```

Reply:
thanks for updating it